### PR TITLE
[BUG-FIX] [Checkout] Change email addressing step

### PR DESCRIPTION
**Pull request by SirDomin**
| Q               | A
| --------------- | -----
| Branch?         | 1.8
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | not sure
| Deprecations?   | no
| Related tickets | 
| License         | MIT

### CODE REVIEW

File: features/checkout/ability_to_change_email_during_checkout.feature [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        And the store has a product "Mantis blade" priced at "$1200"
+        And the store ships everywhere for free
+        And the store allows paying offline
+        And there is a customer "John Doe" identified by an email "john@example.com" and a password "secret"
+
+    @ui
+    Scenario: Seeing email input when i checkout as guest
+        Given I have product "Mantis blade" in the cart
+        When I complete addressing step with email "john@example.com" and "United States" based billing address
+        Then I go to the addressing step
```
Comment (lchrusciel) [inline features/checkout/ability_to_change_email_during_checkout.feature:18]:
```suggestion
        And I go to the addressing step
```

File: features/checkout/ability_to_change_email_during_checkout.feature [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        And I should see email input
```
Comment (lchrusciel) [inline features/checkout/ability_to_change_email_during_checkout.feature:19]:
```suggestion
        And I change my email to "another@email.com"
        Then I should make an order as "another@email.com"
```

File: features/checkout/ability_to_change_email_during_checkout.feature [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+    @ui
+    Scenario: Hiding email input when checking as logged used
+        Given I am logged in as "john@example.com"
+        And I have product "Mantis blade" in the cart
+        And I should not see email input
```
Comment (lchrusciel) [inline features/checkout/ability_to_change_email_during_checkout.feature:25]: Where?
```suggestion
        When I go to the addressing step
        Then I should not be able to change my email
```

File: features/checkout/ability_to_change_email_during_checkout.feature [hunk 4]
```diff
@@ -0,0 +1,28 @@
+@checkout
+Feature: Changing email during checkout with registered email
+    In order to change email during checkout
+    As a customer
```
Comment (GSadee) [inline features/checkout/ability_to_change_email_during_checkout.feature:4]:
```suggestion
    As a Customer
```

File: features/checkout/ability_to_change_email_during_checkout.feature [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+    Background:
+        Given the store operates on a single channel in "United States"
+        And the store has a product "Mantis blade" priced at "$1200"
+        And the store ships everywhere for free
+        And the store allows paying offline
+        And there is a customer "John Doe" identified by an email "john@example.com" and a password "secret"
+
+    @ui
+    Scenario: Seeing email input when i checkout as guest
```
Comment (GSadee) [inline features/checkout/ability_to_change_email_during_checkout.feature:15]:
```suggestion
    Scenario: Being able to change the email when I checking out as a guest
```
?

File: features/checkout/ability_to_change_email_during_checkout.feature [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+        Given I have product "Mantis blade" in the cart
+        When I complete addressing step with email "john@example.com" and "United States" based billing address
+        And I go to the addressing step
```
Comment (GSadee) [inline features/checkout/ability_to_change_email_during_checkout.feature:18]:
```suggestion
        And I go back to the addressing step
```

File: features/checkout/ability_to_change_email_during_checkout.feature [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+        And I complete addressing step with email "new-email@example.com" and "United States" based billing address
+        And I should be checking out as "new-email@example.com"
```
Comment (GSadee) [inline features/checkout/ability_to_change_email_during_checkout.feature:20]:
```suggestion
        Then I should be checking out as "new-email@example.com"
```

File: features/checkout/ability_to_change_email_during_checkout.feature [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+    @ui
+    Scenario: Hiding email input when checking as logged used
```
Comment (GSadee) [inline features/checkout/ability_to_change_email_during_checkout.feature:23]:
```suggestion
    Scenario: Being unable to change the email when I checking out as a logged in user
```

File: features/checkout/ability_to_change_email_during_checkout.feature [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+        Given I am logged in as "john@example.com"
+        And I have product "Mantis blade" in the cart
+        When I complete addressing step with "United States" based billing address
```
Comment (GSadee) [inline features/checkout/ability_to_change_email_during_checkout.feature:26]: This step is unnecessary in this scenario. Maybe you could rewrite the scenario to log in during checkout and then check if after going back to addressing page the email change is impossible.

File: src/Sylius/Behat/Context/Ui/Shop/CheckoutContext.php [hunk 10]
```diff
@@ -219,6 +219,22 @@ public function theSubtotalOfItemShouldBe($item, $price)
         Assert::eq($currentPage->getItemSubtotal($item), $price);
     }
 
+    /**
+     * @Then I should see email input
+     */
+    public function iShouldSeeEmailInput(): void
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Ui/Shop/CheckoutContext.php:225]: Do you use this step somewhere?

**Review by lchrusciel — APPROVED** [review]

File: features/checkout/ability_to_change_email_during_checkout.feature [hunk 11]
```diff
@@ -0,0 +1,28 @@
+@checkout
+Feature: Changing email during checkout with registered email
```
Comment (lchrusciel) [inline features/checkout/ability_to_change_email_during_checkout.feature:2]: Can we synchronize the name of the file with the feature name

File: features/checkout/ability_to_change_email_during_checkout.feature [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+    In order to change email during checkout
```
Comment (lchrusciel) [inline features/checkout/ability_to_change_email_during_checkout.feature:3]:
```suggestion
    In order to always check out with the relevant email address
```

File: features/checkout/ability_to_change_email_during_checkout.feature [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+    As a Customer
+    I want to see email input field when im not logged in
```
Comment (lchrusciel) [inline features/checkout/ability_to_change_email_during_checkout.feature:5]:
```suggestion
    I want to be able to change email if I'm not logged in
```

**Comment by lchrusciel** [comment]
Thank you, @SirDomin! :tada:

**Pull request merged**